### DISCUSSION TITLE
Typescript error when trying to call action from getter

### DISCUSSION THREAD

**Discussion by simshaun** [vuejs/pinia#450] [Help and Questions] [answered]
Is this a bug?

```ts
// Contrived example

import { defineStore } from 'pinia'

export const useTestStore = defineStore({
  id: 'test',

  state() {
    return {}
  },

  getters: {
    cssProperties(): string[] {
      const props = []

      for (const prop of ['foo', 'bar']) {
        props.push(this.getProp(prop))
      }

      return props
    },
  },

  actions: {
    getProp(prop: string): string {
      return prop
    },
  },
})
```

![image](https://user-images.githubusercontent.com/379519/115973755-70d03b80-a525-11eb-9b67-19c22e41ecda.png)

**Comment by posva** [comment, marked as answer]
This is rather intentionally left out because getters being computed properties they should not have side effects and therefore not call actions. It will work because `this` is the store but it isn't typed with the existing actions.

If you want to extract code into a function, it should rather be a regular function that takes as an argument whatever is needed to compute a value

**Reply by simshaun** [reply]
Gotcha. Extracted function it is. Thanks

**Reply by zh-pan** [reply]
But if this is the case, when I need get some 'like button' object, and when anyone button clicked, need call some action.
